controllers/base_controller.js

```diff
@@ -324,31 +324,54 @@ module.exports = function BaseControllerModule(pb) {
         Object.keys(this.siteObj.supportedLocales).forEach(function(locale) {
             var path = self.req.url;
             var isLocalizedPath = !!self.pathVars.locale && path.indexOf(self.pathVars.locale) >= 0;
-            if (self.ls.language === locale && !isLocalizedPath) {
-                //skip current language.  We don't need to list it as an alternate
-                return;
-            }
             var relationship = self.ls.language === locale ? 'canonical' : 'alternate';
-
             var urlOpts = {
                 hostname: self.hostname,
                 locale: undefined
             };
-            if (self.ls.language === locale) {
-                path = path.replace(locale + '/', '').replace(locale, '');
-            }
-            else if (isLocalizedPath) {
-                path = path.replace(self.pathVars.locale, locale);
-            }
-            else {
-                urlOpts.locale = locale;
+            if (self.siteObj.origin && self.siteObj.prefix) {
```

Comment:
The Origin and Prefix was added to support reverse proxies and are currently only being used by Allegis, so these changes should be good to go. I think having a setting on the Site config page would be a better solution to identifying this. And this will enable the change for Aerotek, EASi, Aston Carter, Allegis Global Solutions, TekSystems, but not Allegis Group, as it has a site origin but not the prefix.

Reply:
I understand your point, but my concern is in the increase of settings that are really just for a single client (Allegis). This has happened in the past. Also, all all this work would not really be needed if the reverse proxy did a proper work to replace links (like TEKSystems currently does).

Having said that, I actually have some concerns these changes could potentially affect the single successful reverse proxy (TEKSystems), since we really don't control or know what is going to be the final result of the rewriting rules of such reverse proxy.

We are still missing a second approval. I know you mention you are OK with the changes but I think we still need that approval cc @emontoyacb @sindydanely

Reply:
Okay then based on that, I'm closing this conversation.
Just a reminder that the change of package version is to be done.
CC: @emontoyacb @sindydanely

Reply:
@CB-Abhinav-S I think that we should be good without having a new configuration setting just for this. The reasoning is that if a client already have redirection settings active this metadata will be incorrectly pointing to a different site (the tn domain) so, with this change, we are correctintg that by making the metadata match the redirect url.